test/chart/BarChart.spec.tsx

```diff
@@ -265,7 +271,7 @@ describe('<BarChart />', () => {
   });
 
   test('Render customized shapem when shape is set to be a function', () => {
-    const renderShape = (props: any) => {
+    const renderShape = (props: RectangleProps & DataType) => {
```

Comment:
This is very similar to the implementation provided by the user in the reproduction linked in the issue. 

This also would throw a type error prior to the change in this PR.